orca/src/ds.rs

```diff
@@ -0,0 +1,414 @@
+                                format!(r#"(targetattr="cn || member || gidNumber || description || objectClass")(targetfilter="(objectClass=groupOfNames)")(version 3.0; acl "Enable group_admin to manage groups"; allow (write,add, delete)(groupdn="ldap:///cn=priv_group_manage,{}");)"#, self.ldap.basedn),
+                                r#"(targetattr="objectClass || description || nsUniqueId || uid || displayName || loginShell || uidNumber || gidNumber || gecos || homeDirectory || cn || memberOf || mail || nsSshPublicKey || nsAccountLock || userCertificate")(targetfilter="(objectClass=posixaccount)")(version 3.0; acl "Enable anyone user read"; allow (read, search, compare)(userdn="ldap:///anyone");)"#.to_string(),
+                                r#"(targetattr="displayName || legalName || userPassword || nsSshPublicKey")(version 3.0; acl "Enable self partial modify"; allow (write)(userdn="ldap:///self");)"#.to_string(),
+                                format!(r#"(targetattr="uid || description || displayName || loginShell || uidNumber || gidNumber || gecos || homeDirectory || cn || memberOf || mail || legalName || telephoneNumber || mobile")(targetfilter="(&(objectClass=nsPerson)(objectClass=nsAccount))")(version 3.0; acl "Enable user admin create"; allow (write, add, delete, read)(groupdn="ldap:///cn=priv_account_manage,{}");)"#, self.ldap.basedn),
+                            ]
+                        }
+                    }
+                ]
+            };
+        self.ldap.modify(acimod).await?;
+
+        // Add members as needed.
+        let mut priv_account = Vec::new();
+        let mut priv_group = Vec::new();
+
+        for (id, list) in access.iter() {
+            // get the users name.
+            let account = all_entities.get(id).unwrap();
+
+            let need_account = list
+                .iter()
+                .filter(|v| matches!(v, EntityType::Account(_)))
+                .count()
+                == 0;
+            let need_group = list
+                .iter()
+                .filter(|v| matches!(v, EntityType::Group(_)))
+                .count()
+                == 0;
+
+            if need_account {
+                priv_account.push(account.get_ds_ldap_dn(&self.ldap.basedn))
+            }
+            if need_group {
+                priv_group.push(account.get_ds_ldap_dn(&self.ldap.basedn))
+            }
+        }
+
+        // Sort and dedup
+        priv_account.sort_unstable();
+        priv_group.sort_unstable();
+        priv_account.dedup();
+        priv_group.dedup();
+        // Do the mod in one pass.
+        info!("Setting up cn=priv_group_manage");
+        let req = LdapModifyRequest {
+            dn: format!("cn=priv_group_manage,{}", self.ldap.basedn),
+            changes: vec![LdapModify {
+                operation: LdapModifyType::Delete,
```

Comment:
This is deleting the priv_group?

Reply:
No, it's deleting the members of the priv_group (its an ldap modify so it's changing attributes, where if it was an LdapDelete it would be deleting the priv_group)